### ISSUE TITLE
incorrect positioning of tooltips

### ISSUE THREAD

**Issue by mghohoo** [twbs/bootstrap#19367]
![huaban 1](https://cloud.githubusercontent.com/assets/1341859/13454982/a48056b8-e095-11e5-9be0-e85fadfce64a.png)

Bootstrap v3.3.6
if placement=left or right
and the toptip's target is not in the view when tooltip('show') is executed.

page source attached

[page.txt.txt](https://github.com/twbs/bootstrap/files/154329/page.txt.txt)

**Comment by cvrebert** [comment]
> page source attached

Please use JS Bin / JS Fiddle / CodePen / etc. instead in the future.

JS Bin of your example: http://jsbin.com/romuho/1/edit?html,output

**Comment by twbs-lmvtfy** [comment]
Hi @cvrebert!

You appear to have posted a live example (http://jsbin.com/romuho/1/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
WontFix per https://github.com/twbs/bootstrap/issues/15590#issuecomment-113321235

**Comment by iatek** [comment]
I was able to workaround this using 'viewport' and 'container' options. Also, put the buttons on a `.container`..

`data-container="body" data-viewport=".container" `

```
<div class="container">
    <button style="margin:50px" type="button" class="btn btn-default tips" data-container="body" data-viewport=".container" data-trigger="manual" data-placement="left" title="Tooltip on right">Tooltip on left</button>
    <button style="margin:50px" type="button" class="btn btn-default tips" data-container="body" data-viewport=".container" data-trigger="manual" data-placement="right" title="Tooltip on right">Tooltip on right</button>
    <button style="margin:50px" type="button" class="btn btn-default tips" data-trigger="manual" data-placement="top" title="Tooltip on right">Tooltip on top</button>
    <button style="margin:50px" type="button" class="btn btn-default tips" data-trigger="manual" data-placement="bottom" title="Tooltip on right">Tooltip on bottom</button>

    <div style="height:900px"></div>
    <button type="button" id="trigger">click it</button>
</div>
```

http://codeply.com/go/SyYXzrVuRm
